### PR TITLE
Post-ICHEP Cut-based electron ID and HLT-safe electron ID

### PR DESCRIPTION
**Pull request by emanueledimarco**
- implement SPRING16 cut-based electron ID
- implement the HLT-safe electron ID
- add the output of HLT-safe ID to lepton objects in the ntuples

### CODE REVIEW

**Comment by mariadalfonso** [comment]
@emanueledimarco do you have also the SPRING16 photonID ?

**Comment by emanueledimarco** [comment]
@mariadalfonso to my knowledge it is not yet finalised. The last one recommended on the [twiki](https://twiki.cern.ch/twiki/bin/view/CMS/CutBasedPhotonIdentificationRun2#Photon_ID_Working_Points_WP_defi) is the spring15 one, which is the one we already have implemented.

File: PhysicsTools/Heppy/python/physicsobjects/Electron.py [hunk 1]
```diff
@@ -51,10 +54,10 @@ def cutBasedId(self, wp, showerShapes="auto"):
             showerShapes = "full5x5"
             wp = wp.replace("_full5x5","")
         elif showerShapes == "auto":
-            if "POG_CSA14_25ns_v1" in wp or "POG_CSA14_50ns_v1" in wp or "POG_PHYS14_25ns_v1" in wp or "POG_PHYS14_25ns_v1_ConvVeto" in wp or "POG_PHYS14_25ns_v1_ConvVetoDxyDz" in wp or "POG_PHYS14_25ns_v2" in wp or "POG_PHYS14_25ns_v2_ConvVeto" in wp or "POG_PHYS14_25ns_v2_ConvVetoDxyDz" in wp or "POG_SPRING15_25ns_v1" in wp or "POG_SPRING15_25ns_v1_ConvVeto" in wp or "POG_SPRING15_25ns_v1_ConvVetoDxyDz" in wp or "POG_SPRING15_50ns_v2" in wp or "POG_SPRING15_50ns_v2_ConvVeto" in wp or "POG_SPRING15_50ns_v2_ConvVetoDxyDz" in wp:
+            if "POG_CSA14_25ns_v1" in wp or "POG_CSA14_50ns_v1" in wp or "POG_PHYS14_25ns_v1" in wp or "POG_PHYS14_25ns_v1_ConvVeto" in wp or "POG_PHYS14_25ns_v1_ConvVetoDxyDz" in wp or "POG_PHYS14_25ns_v2" in wp or "POG_PHYS14_25ns_v2_ConvVeto" in wp or "POG_PHYS14_25ns_v2_ConvVetoDxyDz" in wp or "POG_SPRING15_25ns_v1" in wp or "POG_SPRING15_50ns_v2" in wp or "POG_SPRING16_25ns_v1" in wp:
                 showerShapes = "full5x5"
```
Comment (gpetruc) [inline PhysicsTools/Heppy/python/physicsobjects/Electron.py]: Maybe we should revert the logic and say that showerShapes is always full5x5 unless requested.
I'm not aware of anything using ZS cluster shapes.

Comment (emanueledimarco) [reply PhysicsTools/Heppy/python/physicsobjects/Electron.py]: - Done. Note that I changed it only for the cut-based electron IDs (did not touched MVAs). 
- This changes also the name of the ntuple variable "eleCutId2012_full5x5" => "eleCutId2012"
- People using LeptonAnalyzer should remove the "_full5x5" suffix in the e-ID name

**Comment by mariadalfonso** [comment]
@gpetruc 
since is "tested and ok" can be merged ?

**Comment by gpetruc** [comment]
Sorry, I wanted to re-test it after the last commit.
Will test now and then merge if the test is ok

**Pull request merged**